lambda/events_id/get/index.js

```diff
@@ -2,104 +2,127 @@
+        body: body
     };
+    
+    // validate input parameter as uuid version 5, when fails a 400 response must be given
+    if(!isUUID.anyNonNil(eventId)) {
+    	console.log("ERROR: uuid in not uuid/v4");
+    	response.statusCode = 400;
+		callback(JSON.stringify(response), null);
+	
+    }
+    else {
+
+        // parameters for DynamoDB request. 
+        const params = {
+            TableName: 'ak-prod-events-dynamo',
+            KeyConditionExpression: "#event_id = :event_id",
+            // Specify which attribute to search for
+            ExpressionAttributeNames:{
+                "#event_id": "event_id"
+            },
+            // Specify which attribute value {event id} to search for
+            ExpressionAttributeValues: {
+                ":event_id": eventId
+            },
+            // ProjectionExpression is for the request attributes of events in table 
+            ProjectionExpression: "event_id, user_name, activity_type, description, organization_name, location_name, location_address, contact_name, contact_phone, contact_email, start_date_time, end_date_time, frequency, cost, picture_url, min_age, max_age, disability_types, show_status, approver"
+        }//end of params
+    
+        try {
+            // Creating variable {data} to store the response from DynamoDB using the params created above
+            // documentClient.query() will search the entire table based on the parameters given.
+            const data = await documentClient.query(params).promise();
+            response.body = data;
+            console.log(JSON.stringify(response));
+            // Checking if the query returned with an event or found no matches
+            if(response.body.Count == 0) {
+                //setting responseBody and statusCode on no results returned from DynamoDB
+                response.Body = 'Event not found';
+                response.statusCode = 404;
+            
+                callback(response, null)
+                // Log for no event found
+                console.log("Query did not find a matching event with the event_id provided. Status code: ", statusCode);
+            } else {
+                //setting responseBody and statusCode on successful return of event from DynamoDB
+                response.body = response.body.Items;
+                response.statusCode = 200;
+                
+                callback(null, response.body)
+                // Log for event found
+                console.log("Event found, status code: ", statusCode);
+            }
+            
+        } catch(err) {
+            response.body = 'Failure to connect or retrieve from the database';
+            response.statusCode = 500;
+            
+            // Log for error with DynamoDB query
+            console.error("Unsuccessful connection to DynamoDB. Status code: " + statusCode);
+            console.error("Error: " + err);
+            callback(JSON.stringify(response), null)
```

Comment:
This is fine, but API Gateway needs to be configured to handle the response accordingly, based on the string 'statusCode = 500'